pmd-apex/src/main/resources/category/apex/codestyle.xml

```diff
@@ -95,6 +118,30 @@ if (foo) {  // preferred approach
         </example>
     </rule>
 
+    <rule name="FieldRegexNamingConventions"
```

Comment:
I think it would be nice to be consistent with the java rules and name those rules eg "FieldNamingConventions", without the "Regex". The names are already long too. @rsoesemann @jeffhube Wdyt?

Reply:
I fully agree with @adangel and @oowekyala that we should replace the old rules and be consistent with Java as much as possible. @jeffhube can you please incorporate that?

Reply:
@oowekyala The issue here was the existing ClassNamingConventions and MethodNamingConventions rules. At minimum, ClassNamingConventions will not flag class names containing underscore while ClassRegexNamingConventions will, and MethodNamingConventions has a skipTestMethodUnderscores property that is not needed in MethodRegexNamingConventions.

Reply:
@jeffhube I think the differences are very small and not enough to justify creating new rules, especially since the similar names may be confusing to users. The `skipTestMethodUnderscores` property may be deprecated, and the default for ClassNamingConventions can stay the same for compatibility. The default could also be updated, though this is mostly for you Apex experts to consider, and should be clearly documented

(Deprecating a property is done by prefixing its description with `"deprecated!"`)

Reply:
@oowekyala @rsoesemann 

- I've renamed the rules to no longer contain the Regex suffix, and ClassRegexNamingConventions and MethodRegexNamingConventions have been merged into the existing rules.
- In order to avoid breaking changes, The new regexes for ClassNamingConventions will now allow underscores by default.
- The skipTestMethodUnderscores property for MethodNamingConventionsRule is now deprecated, but remains functional. When set, we will use a different regex for test methods that allows underscores.
- VariableNamingConventionsRule is also now deprecated, and I have indicated which rules replace it in the documentation, though I did not note which version it will be removed in.
- I set Code Climate properties in the new rules to be consistent with the existing rules.

Reply:
:+1: